app/models/miq_request.rb

```diff
@@ -434,7 +437,13 @@ def execute
       :tracking_label => tracking_label_id,
       :msg_timeout    => 3600,
       :deliver_on     => deliver_on
-    )
+    }
+    user = User.current_user
```

Comment:
@lfu How does the create_request_tasks method use the User.current_user.
From the way its coded it seems like it would be OK for the User.current_user to be set or missing.
If create_request_tasks truly needs the q_options[:user_id] and the q_options[:group_id] then we should be throwing an exception if the User.current_user is missing.

Reply:
@mkanoor It is used to send event [```request_starting```](https://github.com/ManageIQ/manageiq/blob/0c77dd4c4e7d7f8690a58cdc85034142c924fad7/app/models/miq_request.rb#L447) to automate.

Without this setting, ```MiqAeEvent.call_automate``` would rely on ```obj.tenant_identity``` to get the user.

All these setting regarding user is for raising event to automate which requires a user. ```MiqAeEvent.call_automate``` would throw an error if a user is missing.